File: Form1.cs
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.Commands;
using DevExpress.XtraRichEdit.Services;
using System.Text;
using DevExpress.XtraRichEdit.API.Native;


namespace RichEditCustomCopyPaste {
    public partial class Form1 : Form {
        
        public Form1() {
            InitializeComponent();

            CustomRichEditCommandFactoryService commandFactory = new CustomRichEditCommandFactoryService(richEditControl1, richEditControl1.GetService<IRichEditCommandFactoryService>());
            richEditControl1.RemoveService(typeof(IRichEditCommandFactoryService));
            richEditControl1.AddService(typeof(IRichEditCommandFactoryService), commandFactory);
            richEditControl1.LoadDocument("testDocument.docx");
        }
    }

    public class CustomRichEditCommandFactoryService : IRichEditCommandFactoryService {
        readonly IRichEditCommandFactoryService service;
        readonly RichEditControl control;

        public CustomRichEditCommandFactoryService(RichEditControl control, IRichEditCommandFactoryService service) {
            DevExpress.Utils.Guard.ArgumentNotNull(control, "control");
            DevExpress.Utils.Guard.ArgumentNotNull(service, "service");
            this.control = control;
            this.service = service;
        }

        public RichEditCommand CreateCommand(RichEditCommandId id) {
            if (id == RichEditCommandId.CopySelection )
                return new CustomCopySelectionCommand(control);

            return service.CreateCommand(id);
        }
    }

   
}

File: VB/Form1.vb
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.Commands
Imports DevExpress.XtraRichEdit.Services
Imports System.Text
Imports DevExpress.XtraRichEdit.API.Native


Namespace RichEditCustomCopyPaste
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()

            Dim commandFactory As New CustomRichEditCommandFactoryService(richEditControl1, richEditControl1.GetService(Of IRichEditCommandFactoryService)())
            richEditControl1.RemoveService(GetType(IRichEditCommandFactoryService))
            richEditControl1.AddService(GetType(IRichEditCommandFactoryService), commandFactory)
            richEditControl1.LoadDocument("testDocument.docx")
        End Sub
    End Class

    Public Class CustomRichEditCommandFactoryService
        Implements IRichEditCommandFactoryService

        Private ReadOnly service As IRichEditCommandFactoryService
        Private ReadOnly control As RichEditControl

        Public Sub New(ByVal control As RichEditControl, ByVal service As IRichEditCommandFactoryService)
            DevExpress.Utils.Guard.ArgumentNotNull(control, "control")
            DevExpress.Utils.Guard.ArgumentNotNull(service, "service")
            Me.control = control
            Me.service = service
        End Sub

        Public Function CreateCommand(ByVal id As RichEditCommandId) As RichEditCommand Implements IRichEditCommandFactoryService.CreateCommand
            If id = RichEditCommandId.CopySelection Then
                Return New CustomCopySelectionCommand(control)
            End If

            Return service.CreateCommand(id)
        End Function
    End Class


End Namespace